lib/validate.js

```diff
@@ -39,6 +59,22 @@ module.exports = function validateMiddleware (opts) {
     }
   });
 
+  assert(
+    isObject(opts.joiValidateOptions),
+    'opts.joiValidateOptions must be an Object if provided'
+  );
+
+  ['update', 'create', 'list'].forEach(function (type) {
+    assert(
+      opts.joiValidateOptions[type] === undefined ||
+      opts.joiValidateOptions[type] === null ||
+      opts.joiValidateOptions[type] !== null &&
+      typeof opts.joiValidateOptions[type] === 'object' &&
```

Comment:
I think this can be simplified to _isObject_ and the Array piece can be removed.